### PR TITLE
fix: correct namespace when using a file

### PR DESCRIPTION
**Pull request by Duologic**
This is an edge case, `namespace` should match the full execution path if `path` is a file that is not `main.jsonnet`.

### CODE REVIEW

File: pkg/tanka/load_test.go
Comment (sh0rez) [inline pkg/tanka/load_test.go:126]: This creates inconsistency, where I could receive a different namespace depending on whether I invoke my environment as `./cases/withenv` or `./cases/withenv/main.jsonnet`. This would trip up `prune` once `pruneMark` is merged.

To be consistent, we should probably always include the filename

**Review by sh0rez — CHANGES_REQUESTED** [review]

File: pkg/tanka/load_test.go
Comment (Duologic) [reply pkg/tanka/load_test.go:126]: > To be consistent, we should probably always include the filename
Shall I change that here?

Comment (Duologic) [reply pkg/tanka/load_test.go:126]: see a990c8d

File: pkg/tanka/inline.go
Comment (sh0rez) [inline pkg/tanka/inline.go:53]:
```suggestion
	file, err := jpath.Entrypoint(path)
```

`jpath.Entrypoint` does not return a baseDirectory, but instead the absolute filename of the entrypoint

File: pkg/tanka/inline.go
Comment (sh0rez) [inline pkg/tanka/inline.go:48]: use `jpath.FindRoot` instead

File: pkg/tanka/load_test.go
Comment (sh0rez) [inline pkg/tanka/load_test.go:96]: the `StaticLoader` needs modification as well, it should be consistent with this behavior here

**Review by sh0rez — CHANGES_REQUESTED** [review]

File: pkg/tanka/load_test.go
Comment (Duologic) [reply pkg/tanka/load_test.go:96]: That has bigger implications as it uses `ParseDir`, I made a commit, see what you make of it.

File: pkg/spec/spec.go
Comment (sh0rez) [inline pkg/spec/spec.go:23]: You could get around introducing another argument here, by using `jpath.FsDir` (badly named, ik)
https://github.com/grafana/tanka/blob/22de89f3176d3f16e6055dd4329239a1cff791f9/pkg/jsonnet/jpath/dirs.go#L109

That function takes a path and trims the filename from the end, if it exists.
You could then pass the full filename as an argument, and trim the file part away for legacy compat

**Review by sh0rez — CHANGES_REQUESTED** [review]

**Comment by Duologic** [comment]
(wait, I broke test)

File: pkg/spec/spec.go
Comment (sh0rez) [inline pkg/spec/spec.go:24]: Exactly like that!

It is technically not even required to pass baseDir, as it could be constructed based on `namespace` (better call that entrypoint or path?), but that might be for another refactor

File: pkg/spec/spec.go
Comment (Duologic) [reply pkg/spec/spec.go]: Can you elaborate? It looks like we need to push all the jpath logic into ParseDir. It would end up having `ParseDir(path string)` as a signature. Is that acceptable?

File: pkg/spec/spec.go
Comment (Duologic) [reply pkg/spec/spec.go]: PTAL

File: pkg/spec/spec.go
Comment (sh0rez) [inline pkg/spec/spec.go:46]: This error checking is not necessary, as `jpath.Dirs` is guaranteed to give directories. Before however, you could pass arbitrary values here

**Review by sh0rez — APPROVED** [review]
Approving with one nit

File: pkg/spec/spec.go
Comment (Duologic) [reply pkg/spec/spec.go:46]: I wondered about that.

**Pull request merged**
